provision/README.md

```diff
@@ -1,43 +1,142 @@
+Configuration can be specified in [config.toml](configs/config.toml). Config file can specify all the settings that environment variables can configure and in addition
+`/mapping` endpoint provision layout can be configured.
+
+In `config.toml` we can enlist array of things and channels that we want to create and make connections between them which we call provision layout.
+Metadata can be whatever suits your needs except that at least one thing needs to have `externalID` (which is populated with value from [request](#example)).
+For channels metadata `type` is reserved for `control` and `data` which we use with [Agent](https://github.com/mainflux/agent).
+
+Example below
+```
+[[things]]
+  name = "thing"
+
+  [things.metadata]
+    externalID = "xxxxxx"
+
+
+[[channels]]
+  name = "control-channel"
+
+  [channels.metadata]
+    type = "control"
+
+[[channels]]
+  name = "data-channel"
+
+  [channels.metadata]
+    type = "data"
+
+[[channels]]
+  name = "export-channel"
+
+  [channels.metadata]
+    type = "data"
+```
+
+## Authentication
+In order to create necessary entities provision service needs to authenticate against Mainflux api. To provide authentication credentials to the provision service you can pass it in an environment variable or in a config file as Mainflux user and password or as api token (that can be issued on users/keys endpoint). Additionally users or api token can be passed in Authorization header, this authentication takes precedence over others.
+* MFUser, MFPass  in config or environment
+* MFAPiKEy in config or environment
+* Authorization: Token|ApiKey
+
+
+## Running
+
+Provision service can be run as a standalone or in docker composition as addon to the core docker composition.
+```
+docker-compose -f docker/addons/provision/docker-compose.yml up
+```
+or
+```
+MF_PROVISION_BS_SVC_URL=http://localhost:8202/things MF_PROVISION_THINGS_LOCATION=http://localhost:8182 MF_PROVISION_USERS_LOCATION=http://localhost:8180 MF_PROVISION_CONFIG_FILE=docker/addons/provision/configs/config.toml build/mainflux-provision
+```
+
+For the case that credentials or api token is passed in configuration
+```
+curl -s -S  -X POST  http://localhost:8888/mapping  -H 'Content-Type: application/json' -d '{ "external_id" : "33:52:77:99:43", "external_key":"223334fw2" }'
+```
+
+In the case that provision service is not deployed with credential or you want to use user other than default one being set in environment ( or config file)
+```
+curl -s -S  -X POST  http://localhost:8091/mapping -H "Authorization: $TOK" -H 'Content-Type: application/json'   -d '{ "external_id" : "02:42:fE:65:D3:23", "external_key":"223334fw2" }'
+```
+
+Response contains created things and channels and certificates if any.
+```
+{
+  "things": [
+    {
+      "ID": "c22b0c0f-8c03-40da-a06b-37ed3a72c8d1",
+      "Owner": "",
+      "Name": "thing",
+      "Key": "007cce56-e0eb-40d6-b2b9-ed348a97d1eb",
+      "Metadata": {
+        "externalID": "33:52:79:C3:43"
```

Comment:
`id`, `owner`, `name` etc etc

Reply:
this is result of marshalling things.Thing should we add json tags to that structure maybe?

Reply:
Yes, you need JSON tags. We don't use camelCase for JSON keys

Reply:
Maybe you can do something like here: https://github.com/mainflux/mainflux/blob/master/things/api/things/http/responses.go#L39

Reply:
It should be ok now with this: https://github.com/mainflux/mainflux/pull/1146